src/gui/sdlmain.cpp

```diff
@@ -3243,6 +3254,10 @@ static SDL_Rect calc_viewport(const int win_w, const int win_h)
 		// Calculate the final viewport.
 		view_w = sdl.draw.width * integer_scale_factor;
 		view_h = iround(sdl.draw.height * integer_scale_factor * pixel_aspect);
+
+		if (!is_within_bounds(view_w, view_h)) {
+			std::tie(view_w, view_h) = calculate_bounded_dims();
+		}
```

Comment:
`integer_scale_factor < 1` right above would indicate that the image doesn't fit into window, so this can be moved there instead.

Reply:
Yeah ok, I can do that, plus add some comments.